editor/components/block-list/block.js

```diff
@@ -680,12 +680,12 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps ) => {
 		onMerge( ...args ) {
 			mergeBlocks( ...args );
 		},
-		onReplace( blocks ) {
+		onReplace( blocks, blockToSelectUid ) {
 			const { layout } = ownProps;
 			blocks = castArray( blocks ).map( ( block ) => (
-				cloneBlock( block, { layout } )
+				{ ...block, layout }
 			) );
-			replaceBlocks( [ ownProps.uid ], blocks );
+			replaceBlocks( [ ownProps.uid ], blocks, blockToSelectUid );
```

Comment:
Why don't we just call `selectBlock` or something like that right after? It seems weird to tie the replacement of blocks with selection?

Reply:
Now the action object contains the property selectBlock, unfortunately, I was not able to use it in the parameter name because we already have a function with that name.

Reply:
I think you miss understood my remark here. For me we shouldn't add this new parameter here and just call another action right after `selectBlock( blockToSelectUid )`.

Reply:
Hi @youknowriad I missed your suggestion at first but now I updated the code to follow it :)